src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx

```diff
@@ -437,4 +442,154 @@ describe('VAOS <AppointmentsPageV2>', () => {
   });
+
+  it('should navigate to confirmed appointments detail page', async () => {
+    // VA appointment id from confirmed_va.json
+    const url = '/va/var21cdc6741c00ac67b6cbf6b972d084c1';
+
+    const appointment = getVAAppointmentMock();
+    appointment.attributes = {
+      ...appointment.attributes,
+      startDate: moment().format(),
+      clinicId: '308',
+      clinicFriendlyName: "Jennie's Lab",
+      facilityId: '983',
+      sta6aid: '983GC',
+      communityCare: false,
+      vdsAppointments: [
+        {
+          bookingNote: 'New issue: ASAP',
+          appointmentLength: '60',
+          appointmentTime: '2021-12-07T16:00:00Z',
+          clinic: {
+            name: 'CHY OPT VAR1',
+            askForCheckIn: false,
+            facilityCode: '983',
+          },
+          type: 'REGULAR',
+          currentStatus: 'NO ACTION TAKEN/TODAY',
+        },
+      ],
+      vvsAppointments: [],
+    };
+
+    mockAppointmentInfo({
+      va: [appointment],
+      cc: [],
+      requests: [],
+      isHomepageRefresh: true,
+    });
+
+    const facility = {
+      id: 'vha_442GC',
+      attributes: {
+        ...getVAFacilityMock().attributes,
+        type: 'facility',
+        address: {
+          mailing: {},
+          physical: {
+            zip: '80526-8108',
+            city: 'Fort Collins',
+            state: 'CO',
+            address1: '2509 Research Boulevard',
+            address2: null,
+            address3: null,
+          },
+        },
+        id: 'vha_442GC',
+        name: 'Fort Collins VA Clinic',
+        phone: {
+          fax: '970-407-7440',
+          main: '970-224-1550',
+          pharmacy: '866-420-6337',
+          afterHours: '307-778-7550',
+          patientAdvocate: '307-778-7550 x7517',
+          mentalHealthClinic: '307-778-7349',
+          enrollmentCoordinator: '307-778-7550 x7579',
+        },
+        uniqueId: '442GC',
+      },
+    };
+    mockFacilitiesFetch('vha_442GC', [facility]);
+
+    const screen = renderWithStoreAndRouter(
+      <MemoryRouter initialEntries={['/', url]}>
+        <Route exact path="/" component={AppointmentsPageV2} />
+        <Route
+          exact
+          path="/va/:id"
+          component={ConfirmedAppointmentDetailsPage}
+        />
+      </MemoryRouter>,
+      {
+        initialState,
+      },
+    );
+
+    let detailLinks = await screen.findAllByRole('link', {
+      name: /Detail/i,
+    });
+
+    // Select an appointment details link...
+    let detailLink = detailLinks.find(l => l.getAttribute('href') === url);
+    userEvent.click(detailLink);
+
+    // Verify page content...
+    expect(
+      await screen.findByRole('heading', {
+        level: 1,
+        name: new RegExp(moment().format('dddd, MMMM DD, YYYY'), 'i'),
+      }),
+    ).to.be.ok;
+
+    // NOTE: This 2nd 'await' is needed!!! Seems like we need to wait for
+    // conditionals to process in addition to any async code.
```

Comment:
The facility details are fetched asynchronously, so this page could render once with appointment info but without the facility details. That's why the second await is needed.

Reply:
Cool. It wasn't clear to me what was causing the strange behavior I was seeing.